internal/graph/cached_resolver_test.go

```diff
@@ -0,0 +1,514 @@
+		{
+			name: "request_with_different_contextual_tuple_does_not_return_results_from_cache",
+			subsequentReq: &ResolveCheckRequest{
+				StoreID:              "12",
+				AuthorizationModelID: "33",
+				TupleKey:             tuple.NewTupleKey("document:abc", "reader", "user:XYZ"),
+				ContextualTuples: []*openfgav1.TupleKey{
+					{
+						Object:   "document:xxx",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+				},
+			},
+			setInitialResult: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(result, nil)
+			},
+			setTestExpectations: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(result, nil)
+			},
+		},
+		{
+			name: "response_with_error_not_cached",
+			subsequentReq: &ResolveCheckRequest{
+				StoreID:              "12",
+				AuthorizationModelID: "33",
+				TupleKey:             tuple.NewTupleKey("document:abc", "reader", "user:XYZ"),
+			},
+			setInitialResult: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(nil, fmt.Errorf("Mock error"))
+			},
+			setTestExpectations: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(result, nil)
+			},
+		},
+		{
+			name: "identical_contextual_tuples_return_results_from_cache",
+			initialReq: &ResolveCheckRequest{
+				StoreID:              "12",
+				AuthorizationModelID: "33",
+				TupleKey:             tuple.NewTupleKey("document:abc", "reader", "user:XYZ"),
+				ContextualTuples: []*openfgav1.TupleKey{
+					{
+						Object:   "document:aaa",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+					{
+						Object:   "document:xxx",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+				},
+			},
+			subsequentReq: &ResolveCheckRequest{
+				StoreID:              "12",
+				AuthorizationModelID: "33",
+				TupleKey:             tuple.NewTupleKey("document:abc", "reader", "user:XYZ"),
+				ContextualTuples: []*openfgav1.TupleKey{
+					{
+						Object:   "document:aaa",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+					{
+						Object:   "document:xxx",
+						Relation: "reader",
+						User:     "user:XYZ",
+					},
+				},
+			},
+			setInitialResult: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(1).Return(result, nil)
+			},
+			setTestExpectations: func(mock *MockCheckResolver, request *ResolveCheckRequest) {
+				mock.EXPECT().ResolveCheck(ctx, request).Times(0).Return(result, nil)
+			},
+		},
+		{
+			// Ideally we will have the same order. However, having different order
+			// will not be catastrophic - just result in a cache miss and potentially
+			// duplicate entry
+			name: "different_order_contextual_tuples_does_not_return_results_from_cache",
```

Comment:
I suggested we take a checksum of a the contextual tuples to produce a cache key that is stable independent of order.

If we're going to save that work for a subsequent PR, would you please add an issue to track that work and cross reference it here?

Same contextual tuples but different order should result in a cache hit.